config/initializers/app_artifacts.rb

```diff
@@ -10,12 +10,37 @@
   store.add_artifact(:saml_2024_cert, '/%<env>s/saml2024.crt')
   store.add_artifact(:saml_2024_key, '/%<env>s/saml2024.key.enc')
 
-  store.add_artifact(:oidc_private_key, '/%<env>s/oidc.key') { |k| OpenSSL::PKey::RSA.new(k) }
-  store.add_artifact(:oidc_public_key, '/%<env>s/oidc.pub') { |k| OpenSSL::PKey::RSA.new(k) }
+  store.add_artifact(:oidc_primary_private_key, '/%<env>s/oidc.key') do |k|
+    OpenSSL::PKey::RSA.new(k)
+  end
+  store.add_artifact(:oidc_primary_public_key, '/%<env>s/oidc.pub') do |k|
+    OpenSSL::PKey::RSA.new(k)
+  end
+  store.add_artifact(
+    :oidc_secondary_private_key, '/%<env>s/oidc_secondary.key',
+    allow_missing: true
+  ) do |k|
+    OpenSSL::PKey::RSA.new(k) if !k.nil?
+  end
+  store.add_artifact(
+    :oidc_secondary_public_key, '/%<env>s/oidc_secondary.pub',
+    allow_missing: true
+  ) do |k|
+    OpenSSL::PKey::RSA.new(k) if !k.nil?
+  end
```

Comment:
IMO it would be better to not call the block if the value is nil?

```suggestion
  ) do |k|
    OpenSSL::PKey::RSA.new(k)
  end
  store.add_artifact(
    :oidc_secondary_public_key, '/%<env>s/oidc_secondary.pub',
    allow_missing: true
  ) do |k|
    OpenSSL::PKey::RSA.new(k)
  end
```